src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs

```diff
@@ -338,17 +338,13 @@ public SymbolTable GetMutableCopyOfFunctions()
                 DebugName = DebugName + " (Functions only)",
             };
 
-            s.AddFunctions(_functions);
+            s.AddFunctions(this.Functions);
 
             return s;
         }
 
         internal readonly Dictionary<string, NameLookupInfo> _variables = new Dictionary<string, NameLookupInfo>();
-
-        private protected readonly TexlFunctionSet _functions = new TexlFunctionSet();

```

Comment:
Move the mutable one out of the base class and into derived SymbolTable class. 
This lets the derived class assert guards on it.